secure_fw/core/tfm_core_mem_check.c

```diff
@@ -35,6 +35,28 @@ static enum tfm_status_e has_access_to_region(const void *p, size_t s,
     range_access_allowed_by_mpu =
                           cmse_check_address_range((void *)p, s, flags) != NULL;
 
+    if (!range_access_allowed_by_mpu) {
+    	/*
+    	 * Verification failure may be due to address range crossing
+         * IDAU boundaries. We check if the whole range is part of a
+         * single MPU region and that the region satisfies the given
+         * access permissions.
+    	 */
+        const void *p_addr_base = p;
+        const void *p_addr_last = p + s - 1;
+
+        cmse_address_info_t addr_info_base = cmse_TT((void *)p_addr_base);
+        cmse_address_info_t addr_info_last = cmse_TT((void *)p_addr_last);
```

Comment:
Say the mpu region spans 3 IDAU regions, could the middle IDAU region be configured such (e.g. if the middle region is non-secure) that this new code sets `range_access_allowed_by_mpu` to true below when it should have been false?